src/server/pps/pretty/draw.go

```diff
@@ -0,0 +1,371 @@
+				border += hiddenRow
+				row += hiddenRow
+			} else {
+				border += colorSprint(fmt.Sprintf("%s+%s+", strings.Repeat(" ", spacing), strings.Repeat("-", ro.boxWidth)))
+				row += colorSprint(fmt.Sprintf("%s|%s%s%s|", strings.Repeat(" ", spacing), boxPadLeft, label, boxPadRight))
+			}
+			written += spacing + len(boxPadLeft) + len(label) + len(boxPadRight) + 2
+			for _, u := range v.edges {
+				renderEdges = append(renderEdges, renderEdge{src: v.rowOffset, dest: u.rowOffset})
+			}
+		}
+		picture += fmt.Sprintf("%s\n%s\n%s\n", border, row, border)
+		// print up to `layerVerticalSpace` rows that will contain edge drawings
+		for j := 0; j < ro.edgeHeight; j++ {
+			row := strings.Repeat(" ", maxRowWidth)
+			for _, re := range renderEdges {
+				row = re.render(row, j, ro.edgeHeight)
+			}
+			picture += fmt.Sprint(row)
+			picture += "\n"
+		}
+	}
+	return picture
+}
+
+// renderEdge is used to describe the source and destination of an edge in terms of the x-axis.
+// The number of vertical lines spanned is calculated at each layer
+type renderEdge struct {
+	src  int
+	dest int
+}
+
+func (re renderEdge) distance() int {
+	return abs(re.src - re.dest)
+}
+
+// render sets an edge character in the 'row' string, calculated using the re.src & re.dest (the edge's range along the x-axis),
+// and 'vertDist' (the height of the edge) and 'vertIdx' (how many lines down 'vertDist' we are)
+func (re renderEdge) render(row string, vertIdx, vertDist int) string {
+	setEdgeChar := func(s string, i int, r rune) string {
+		if s[i] == byte(r) {
+			return s
+		}
+		if s[i] == ' ' {
+			return s[:i] + string(r) + s[i+1:]
+		}
+		return s[:i] + "+" + s[i+1:] // set the coordinate to "+" if there's an edge crossing
+	}
+	if re.src == re.dest {
+		return setEdgeChar(row, re.src, '|')
+	}
+	const srcEdgeCenterOffset = 1                        // start drawing a diagonal edge one space away from the center of a node
+	adjustedXDist := re.distance() - srcEdgeCenterOffset // number of horizontal spaces we must fill with edges
+	// vertical line
+	if vertDist > adjustedXDist && vertIdx >= adjustedXDist/2 && vertIdx < vertDist-adjustedXDist/2 {
+		return setEdgeChar(row, (re.src+re.dest)/2, '|')
+	}
+	// horizontal line
+	if vertDist < adjustedXDist && vertIdx == vertDist/2 {
+		step := 1
+		if re.src > re.dest {
+			step = -1
+		}
+		diagCoverage := (vertDist / 2) * step
+		tmp := re.src + diagCoverage
+		for tmp != re.dest-diagCoverage-(step*vertDist%2) {
+			tmp += step
+			row = setEdgeChar(row, tmp, '-')
+		}
+		return row
+	}
+	// diagonal line
+	offset := vertIdx + srcEdgeCenterOffset
+	// calculate offset based on distance from the end in case we're on the bottom half of the edge
+	if vertIdx > vertDist/2 {
+		offset = adjustedXDist - (vertDist - offset)
+	}
+	if re.src > re.dest {
+		i := re.src - offset
+		return setEdgeChar(row, i, '/')
+	} else {
+		i := re.src + offset
+		return setEdgeChar(row, i, '\\')
```

Comment:
Similarly
```suggestion
		return setEdgeChar(row, re.src + offset, '\\')
```
I prefer to un-nest this else and just return below the `if` block.